src/hotspot/share/compiler/compilationFailureInfo.cpp

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright (c) 2023, Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 2023, Red Hat, Inc. and/or its affiliates.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ *
+ */
+
+#include "precompiled.hpp"
+
+#if defined(COMPILER1) || defined(COMPILER2)
+
+#ifdef COMPILER1
+#include "c1/c1_Compilation.hpp"
+#endif
+#include "ci/ciEnv.hpp"
+#include "compiler/abstractCompiler.hpp"
+#include "compiler/compilationFailureInfo.hpp"
+#include "compiler/compileTask.hpp"
+#ifdef COMPILER2
+#include "opto/node.hpp"
+#include "opto/compile.hpp"
+#endif
+#include "runtime/os.hpp"
+#include "utilities/ostream.hpp"
+#include "utilities/nativeCallStack.hpp"
+
+static constexpr int skip_frames = 2;
```

Comment:
Is there a specific reason to define this separately and not directly use `_stack(2)` below?

Reply:
Hehe ok. I guess with the IDE parameter hints, it's probably fine to pass a 2 directly. But either way is fine :-)